pkg/generate/app/app.go

```diff
@@ -182,17 +182,34 @@ func (r *ImageRef) NameReference() string {
 }
 */
 
+// nameAndTag returns an image name with a tag. If the tag is blank, it will
+// use the default tag ("latest")
+func nameAndTag(name, tag string) string {
```

Comment:
I feel like I've seen this propogating.  Seems like it makes sense to have `image/api/helpers.go`